### PR TITLE
various follow-ups to socket-bind logic

### PR DESCRIPTION
**Pull request by poettering**
Mostly minor stuff that is a follow-up for #17655.

/cc @wat-ze-hex

### CODE REVIEW

File: src/core/socket-bind.c [hunk 1]
```diff
@@ -25,12 +25,10 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(struct socket_bind_bpf *, socket_bind_bpf_free);
 static int update_rules_map(
                 int map_fd, CGroupSocketBindItem *head) {
         CGroupSocketBindItem *item;
-        uint32_t i = 0;
 
         assert(map_fd >= 0);
 
         LIST_FOREACH(socket_bind_items, item, head) {
-                const uint32_t key = i++;
```
Comment (jkartseva) [inline src/core/socket-bind.c]: This won't compile: `key` is required to add an element to the map:
```
                 if (bpf_map_update_elem(map_fd, &key, &val, BPF_ANY) != 0)
```

**Review by wat-ze-hex — CHANGES_REQUESTED** [review]
Thank you!

LGTM, but this breaks the build and needs to be fixed.

File: src/core/socket-bind.c (hunk 1 above)
Comment (poettering) [reply src/core/socket-bind.c]: uh, gah, in my test env libbpf wasn't installed. let me roll this again. sorry for the confusoin.

**Comment by poettering** [comment]
dropped the non-sensical patch and repushed. This time my test env had all the right tools and it was happy with the results, so let's see if the CI agrees too.

File: src/systemctl/systemctl-show.c [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 #include <sys/mount.h>
 
+#include "af-list.h"
```
Comment (yuwata) [inline src/systemctl/systemctl-show.c:5]: This is added in wrong commit?

File: src/systemctl/systemctl-show.c [hunk 3]
```diff
@@ -1711,14 +1711,15 @@ static int print_property(const char *name, const char *expected_value, sd_bus_m
...
-                        const char *family;
                         int af;
 
                         r = sd_bus_message_enter_container(m, SD_BUS_TYPE_ARRAY, "(iqq)");
                         if (r < 0)
                                 return bus_log_parse_error(r);
                         while ((r = sd_bus_message_read(m, "(iqq)", &af, &nr_ports, &port_min)) > 0) {
-                                family = af == AF_INET ? "ipv4:" : af == AF_INET6 ? "ipv6:" : "";
+                                const char *family =
+                                        af == AF_INET ? "ipv4:" : af == AF_INET6 ? "ipv6:" : "";
```
Comment (yuwata) [inline src/systemctl/systemctl-show.c]: It is better to use `af_to_ipv4_ipv6()`.

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by poettering** [comment]
Force pushed a new version, addressing @yuwata's points

**Comment by keszybz** [comment]
LGTM too.

**Pull request merged**